### PR TITLE
inbox system

### PR DESCRIPTION
**Pull request by chan227**

### CODE REVIEW

**Comment by chan227** [comment]
ready

**Comment by chan227** [comment]
ready

File: app/controllers/conversations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+class ConversationsController < ApplicationController
+  before_action :current_user
+
+  def new; end
+
+  def create
+    recipients = User.where(id: conversation_params[:recipients])
+    conversation = current_user.send_message(recipients, conversation_params[:body], conversation_params[:subject]).conversation
+    flash[:success] = t ".success_mess"
+    redirect_to conversation_path(conversation)
```
Comment (saobv-2223) [inline app/controllers/conversations_controller.rb]: remove last ()

File: app/controllers/conversations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  end
+
+  def show
+    @receipts = conversation.receipts_for(current_user)
+    conversation.mark_as_read(current_user)
```
Comment (saobv-2223) [inline app/controllers/conversations_controller.rb]: same

File: app/controllers/conversations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  end
+
+  def reply
+    current_user.reply_to_conversation(conversation, message_params[:body])
+    flash[:notice] = t ".succ_mess"
+    redirect_to conversation_path(conversation)
```
Comment (saobv-2223) [inline app/controllers/conversations_controller.rb]: same

File: app/helpers/conversations_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module ConversationsHelper
+end
```
Comment (saobv-2223) [inline app/helpers/conversations_helper.rb]: remove file that never used

File: app/views/conversations/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<%= form_for :conversation, url: :conversations, html: { class: "" } do |f| %>
```
Comment (saobv-2223) [inline app/views/conversations/_form.html.erb:1]:
```suggestion
<%= form_for :conversation, url: :conversations, do |f| %>
```

File: app/views/conversations/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      <div class="panel-body">
+        <%= render "mailbox/folders" %>
+      </div>
+    </div>
+  </div>
+
+  <div class="col-md-8">
+    <div class="panel panel-default">
+      <div class="panel-body">
+        <%= render partial: "messages"%>
```
Comment (saobv-2223) [inline app/views/conversations/show.html.erb:29]:
```suggestion
        <%= render partial: "messages", collection: @receipts, as: :receipt %>
```

File: app/views/conversations/_messages.html.erb [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+      <div class="media-body">
+        <h4 class="media-heading">
+          <%= message.sender.name %> <br>
+          <small><b>Subject: </b><%= message.subject %></small><br>
+          <small><b>Date: </b><%=  message.created_at.strftime("%A, %b %d, %Y at %I:%M%p") %></small>
+        </h4>
+        <%= message.body %>
+      </div>
+    </div>
+<% end %>
```
Comment (saobv-2223) [inline app/views/conversations/_messages.html.erb]:
```suggestion
  <% message = receipt.message %>
  <div class="media">
    <div class="media-left">
      <%= image_tag gravatar_for(current_user), size: Settings.users.default_image.list, class: "img-circle" %>
    </div>
    <div class="media-body">
      <h4 class="media-heading">
        <%= message.sender.name %> <br>
        <small><b>Subject: </b><%= message.subject %></small><br>
        <small><b>Date: </b><%=  message.created_at.strftime("%A, %b %d, %Y at %I:%M%p") %></small>
      </h4>
      <%= message.body %>
    </div>
  </div>
```

File: app/views/mailbox/_folders.html.erb [hunk 8]
```diff
@@ -0,0 +1,23 @@
+<ul class="nav nav-pills nav-stacked">
+  <li class="<%= active_page(:inbox) %>">
+    <%= link_to mailbox_inbox_path do  %>
+        <span class="label label-danger pull-right"><%=unread_messages_count %></span>
```
Comment (saobv-2223) [inline app/views/mailbox/_folders.html.erb]:
```suggestion
        <span class="label label-danger pull-right"><%= unread_messages_count %></span>
```

File: app/controllers/conversations_controller.rb (hunk 1 above)
Comment (chan227) [reply app/controllers/conversations_controller.rb]: done

File: app/controllers/conversations_controller.rb (hunk 2 above)
Comment (chan227) [reply app/controllers/conversations_controller.rb]: done

File: app/controllers/conversations_controller.rb (hunk 3 above)
Comment (chan227) [reply app/controllers/conversations_controller.rb]: done

File: app/helpers/conversations_helper.rb (hunk 4 above)
Comment (chan227) [reply app/helpers/conversations_helper.rb]: done

File: app/views/conversations/_form.html.erb (hunk 5 above)
Comment (chan227) [reply app/views/conversations/_form.html.erb:1]: error occurred

File: app/views/conversations/_messages.html.erb (hunk 7 above)
Comment (chan227) [reply app/views/conversations/_messages.html.erb]: error occurred

File: app/views/mailbox/_folders.html.erb (hunk 8 above)
Comment (chan227) [reply app/views/mailbox/_folders.html.erb:4]: error occurred

File: app/views/conversations/show.html.erb (hunk 6 above)
Comment (chan227) [reply app/views/conversations/show.html.erb:29]: error occurred

File: app/views/mailbox/_folders.html.erb (hunk 8 above)
Comment (chan227) [reply app/views/mailbox/_folders.html.erb]: done

File: app/views/conversations/_form.html.erb (hunk 5 above)
Comment (chan227) [reply app/views/conversations/_form.html.erb:1]: image

**Comment by chan227** [comment]
done

**Comment by chan227** [comment]
done

**Comment by chan227** [comment]
done

**Comment by chan227** [comment]
ready

File: app/javascript/packs/conversations.js [hunk 9]
```diff
@@ -0,0 +1,9 @@
+var ready;
+
+ready = function(){
+    $('.chosen-select').chosen({
+        no_results_text: 'No results matched'
+    });
+}
```
Comment (saobv-2223) [inline app/javascript/packs/conversations.js]: check indenting

File: app/views/conversations/_form.html.erb (hunk 5 above)
Comment (saobv-2223) [reply app/views/conversations/_form.html.erb:1]: Ok, sorry for mistake

**Comment by chan227** [comment]
ready

**Pull request merged**